packages/@o3r/pipeline/schematics/ng-add/templates/github/__dot__github/workflows/main.yml

```diff
@@ -0,0 +1,63 @@
+name: Main CI
+
+env:
+  DEFAULT_BRANCH: main
+  DEFAULT_BRANCH_PRERELEASE_NAME: prerelease
+  RELEASE_BRANCH_REGEX: 'release\/(0|[1-9]\d*)\.(0|[1-9]\d*)(\.0-(?:next|prerelease|rc))?$'
+
+on:
+  push:
+    branches:
+      - main
+      - 'release/*'
+  pull_request:
+    branches:
+      - main
+      - 'release/*'
+
+jobs:
+  build:
+    runs-on: <%= runner %>
+    steps:
+      - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+      - uses: ./.github/actions/setup
+      - name: Build
+        run: <%= packageManager %> run build
+
+  unit-tests:
+    runs-on: <%= runner %>
+    steps:
+      - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+      - uses: ./.github/actions/setup
+      - name: Unit tests
+        run: <%= packageManager %> run test
+
+  lint:
+    runs-on: <%= runner %>
+    steps:
+      - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+      - uses: ./.github/actions/setup
+      - name: Lint
+        run: <%= packageManager %> run lint
+
+  version:
+    permissions:
+      contents: write
+    runs-on: <%= runner %>
+    outputs:
+      nextVersionTag: ${{ steps.new-version.outputs.nextVersionTag }}
+    steps:
+      - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+      - name: New version
+        if: github.event_name != 'merge_group'
+        id: new-version
+        uses: AmadeusITGroup/otter/tools/github-actions/new-version@11
```

Comment:
Should be git hash as other actions

Reply:
this one is a bit tricky since it is referring to the same repository as the generator (you cannot get the hash of the commit before committing the code :\)
I think we need to either:
- get the hash somehow at runtime during the generation and use it in the template
- or keep it like this and Renovate will convert it to an hash anyway when it will first run in the app

Reply:
Keeping `11` is not really satisfying neither as it will need to be manually changed every major and it will always really to latest 11 version even if a specific version of Otter is selected.
Without going as far as a hash, a good compromise can be to set the pin version via the `set-version` script/process at pre-publish time